test/scopedvalues.jl

```diff
@@ -0,0 +1,55 @@
+# This file is a part of Julia. License is MIT: https://julialang.org/license
+
+const svar1 = ScopedValue(1)
+
+@testset "errors" begin
+    var = ScopedValue(1)
+    @test_throws MethodError var[] = 2
+    scoped() do
+        @test_throws MethodError var[] = 2
+    end
```

Comment:
another thing we might want to test is trying to create an "empty" ScopedValue
```suggestion
    end
    @test_throws MethodError ScopedValue{Int}()
    @test_throws MethodError ScopedValue()
```